### PR TITLE
Integrate corbado package

### PR DESCRIPTION
**Pull request by nicolaistein**

### CODE REVIEW

File: pubspec.yaml
Comment (incorbador) [inline pubspec.yaml:1]: I think we need to clean up this file:
- remove the default comments
- check which dependencies we really need (dart_ipify, localstorage, corbado_auth look like candidates that we might not need)

File: lib/api/corbado_service.dart [hunk 1]
```diff
@@ -1,118 +1,37 @@
...
 import 'package:flutter/material.dart';
-import 'package:http/http.dart' as http;
+import 'package:passkeys/authenticator/passkey_authenticator.dart';
 
 ///Issues api calls to the corbado backend
 class CorbadoService {
-  final String baseUrlAuth;
-  final String projectID;
+  final String _projectID;
   static String origin = "";
```
Comment (incorbador) [inline lib/api/corbado_service.dart:10]: origin is unused

File: lib/api/corbado_service.dart [hunk 2]
```diff
@@ -1,118 +1,37 @@
...
-        throw Exception(errorStr + error);
-      }
+  final ApiClient _client;
+  late final PasskeyAuthenticator _authenticator;
+
+  CorbadoService(this._projectID)
+      : _client =
+            ApiClient(basePath: 'https://$_projectID.frontendapi.corbado.io') {
+    _authenticator = PasskeyAuthenticator();
+    if (Platform.isAndroid) {
```
Comment (incorbador) [inline lib/api/corbado_service.dart:18]: Do we still need this? I think we no longer whitelist origins (maybe you check with vincent).

File: lib/api/corbado_service.dart [hunk 3]
```diff
@@ -1,118 +1,37 @@
-import 'dart:convert';
+import 'dart:io';
```
Comment (incorbador) [inline lib/api/corbado_service.dart:1]: shall we rename the "api" directory to "services"? To me this seems to be the more common structure.

File: lib/activities/login_activity.dart [hunk 4]
```diff
@@ -1,20 +1,17 @@
...
-import 'package:flutter/services.dart';
 import 'package:oktoast/oktoast.dart';
-import 'dart:async';
-import 'dart:io' show Platform;
 
 class LoginActivity extends StatefulWidget {
   final String projectID;
   final bool emailLinkRedirect;
+  late final CorbadoAuth corbadoAuth;
   late final CorbadoService corbadoSvc;
```
Comment (incorbador) [inline lib/activities/login_activity.dart:15]: corbadoSvc can be removed

File: lib/activities/email_link_confirm_activity.dart [hunk 5]
```diff
@@ -1,8 +1,8 @@
-import 'package:flutter/material.dart';
-import 'package:corbado_demo/api/corbado_service.dart';
 import 'package:corbado_demo/activities/login_activity.dart';
```
Comment (incorbador) [inline lib/activities/email_link_confirm_activity.dart:1]: Hmm if we want to use this as an example for the passkeys package we should remove the email confirmation I think. This is something specific to the Corbado relying party server (so we should move it to an example for the corabdo_auth package).

Comment (incorbador) [reply lib/activities/email_link_confirm_activity.dart:1]: If we remove it there are other files affected as well (e.g. ContentActivity)

**Review by incorbador — CHANGES_REQUESTED** [review]
We need to update the README.md as well.

File: lib/activities/email_link_confirm_activity.dart (hunk 5 above)
Comment (nicolaistein) [reply lib/activities/email_link_confirm_activity.dart:1]: Talked with Vincent, the repo should be an example for the corbado auth package - email confirmation will be integrated into corbado_auth as well, until then we have to use our custom impl for it

**Review by incorbador — APPROVED** [review]

**Pull request merged**